pkg/azure/externalmetrics/monitor_client.go

```diff
@@ -55,23 +55,33 @@ func (c *monitorClient) GetAzureMetric(azMetricRequest AzureExternalMetricReques
 		return AzureExternalMetricResponse{}, err
 	}
 
-	total := extractValue(metricResult)
-
-	klog.V(2).Infof("found metric value: %f", total)
+	value := extractValue(azMetricRequest, metricResult)
 
 	// TODO set Value based on aggregations type
 	return AzureExternalMetricResponse{
-		Total: total,
+		Value: value,
 	}, nil
 }
 
-func extractValue(metricResult insights.Response) float64 {
-	//TODO extract value based on aggregation type
-	//TODO check for nils
+func extractValue(azMetricRequest AzureExternalMetricRequest, metricResult insights.Response) float64 {
 	metricVals := *metricResult.Value
 	Timeseries := *metricVals[0].Timeseries
 	data := *Timeseries[0].Data
-	total := *data[len(data)-1].Total
 
-	return total
+	switch insights.AggregationType(azMetricRequest.Aggregation) {
+	case insights.Average:
+		klog.V(2).Infof("metric type: %s %f", azMetricRequest.Aggregation, *data[len(data)-1].Average)
+		return *data[len(data)-1].Average
+	case insights.Total:
+		klog.V(2).Infof("metric type: %s %f", azMetricRequest.Aggregation, *data[len(data)-1].Total)
+		return *data[len(data)-1].Total
+	case insights.Maximum:
+		klog.V(2).Infof("metric type: %s %f", azMetricRequest.Aggregation, *data[len(data)-1].Maximum)
+		return *data[len(data)-1].Maximum
+	case insights.Minimum:
+		klog.V(2).Infof("metric type: %s %f", azMetricRequest.Aggregation, *data[len(data)-1].Minimum)
+		return *data[len(data)-1].Minimum
+	default:
+		return 0
```

Comment:
Valuable to add a log statement here so we can see that aggregation was not met. 

Is this a failure mode?  Is zero the correct value to return in all cases?  Should we return an error here?